native/common/jp_context.cpp

```diff
@@ -159,6 +171,35 @@ void JPContext::attachJVM(JNIEnv* env)
 	initializeResources(env, false);
 }
 
+std::string getShared() 
+{
+#ifdef WIN32
+	// Windows specific
+	char path[MAX_PATH];
+	HMODULE hm = NULL;
+	if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | 
+		GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+		(LPCSTR) &getShared, &hm) != 0 &&
+		GetModuleFileName(hm, path, sizeof(path)) != 0)
+	{
+		// This is needed when there is no-ascii characters in path
+		char shortPathBuffer[MAX_PATH];
+		GetShortPathName(path, shortPathBuffer, MAX_PATH);
+		return shortPathBuffer;
```

Comment:
The issue here was that GetShortPathName ate the non-ascii characters which allows java to load the library properly.   This was the only pattern that I found to work consistently.   UNICODE doesn't help the problem as the issue is that Java doesn't accept non-ascii anywhere in the path.   It is better to let it mangle to "a~1/b~2/org.jpype.lib" which somehow allows it to function properly.

Reply:
Ok. GetShortPathName is a macro though. If UNICODE is defined it will expand to GetShortPathNameW otherwise GetShortPathNameA.

I understand why this is being done. My point is that it can fail. As long as it fails in a way that we'll know what happened when the user reports a problem, then it's fine.

Reply:
All good.  I let it fall through to the old behavior.